framework/docs/source/how-to-run-flower-on-gcp.rst

```diff
@@ -0,0 +1,633 @@
+
+To install the Google Cloud SDK, we first need to install and configure the ``gcloud``
+CLI:
+
+.. code-block:: bash
+
+    # macOS
+    curl https://sdk.cloud.google.com | bash  # and then follow on-screen prompts
+
+    # macOS (w/ Homebrew)
+    brew install --cask google-cloud-sdk
+
+    # Windows
+    # Download the Windows installer from the Google Cloud SDK page
+    # https://dl.google.com/dl/cloudsdk/channels/rapid/GoogleCloudSDKInstaller.exe
+    # Run the .exe installer and follow the on-screen instructions
+
+    # Once the package is installed, we initialize gcloud as follows:
+    gcloud init  # initialize with gcloud init.
+    source ~/.bashrc  # update PATH
+    gcloud version  # verify installation
+
+.. note::
+
+    For more detailed installation instructions and for installing ``gcloud`` for
+    different operating systems, please take look at the official ``gcloud`` CLI
+    installation page https://cloud.google.com/sdk/docs/install
+
+Once ``gcloud`` is installed we need to install ``kubectl``:
+
+.. code-block:: bash
+
+    gcloud components install kubectl
+    kubectl version --client  # this will show the installed versions of the Client and Kustomize
+
+Now you need to configure ``kubectl`` to point to the GKE cluster you created in the
+previous steps by using the name of the cluster, e.g., ``flower-numpy-example``, and the
+name of the region where the cluster was created:
+
+.. code-block:: bash
+
+    gcloud container clusters get-credentials flower-numpy-example --region us-central1
+
+This will configure the required metadata and fetch the necessary credentials to allow
+your local ``kubectl`` to communicate with the GKE cluster. To verify that ``kubectl``
+was able to connect to the cluster and get the necessary information, you can run the
+following command:
+
+.. code-block:: bash
+
+    kubectl config current-context  # this should return the Kubernetes cluster you connected
+
+.. note::
+
+    For more information on how ``kubectl`` works, please have a look at the following
+    official quick-reference guide:
+    https://kubernetes.io/docs/reference/kubectl/quick-reference/
+
+Create a Google Artifact Repository
+-----------------------------------
+
+The Google Cloud Artifact Registry is a fully managed, scalable, and private service for
+storing and managing software build artifacts and dependencies. Consequently, to run our
+Flower application on the GKE cluster, we need to store the application's specific
+Flower Docker images within the registry, i.e., ``ClientApp`` and ``ServerApp``, which
+we discuss in the next section. This step is crucial as it enables the cluster, and
+subsequently the pods, to download the built Docker images and deploy the necessary
+Flower components.
+
+There are two ways to create the required registry, one through the UI and another
+through the CLI., below we discuss both approaches.
+
+**Create through UI**:
```

Comment:
Given that `gcloud` is installed I wonder if it's enough to keep the `through CLI` instructions. Part of the motivation is related to keeping this guide up to date. The UI might change more often than CLI.